src/iexamine.cpp

```diff
@@ -2280,6 +2280,179 @@ void iexamine::tree_hickory(player &p, const tripoint &examp)
     none( p, examp );
 }
 
+void iexamine::tree_maple(player &p, const tripoint &examp)
+{
+    if( !p.has_items_with_quality( "DRILL", 1, 1 ) ) {
+        add_msg( m_info, _( "You need a tool to drill the crust to tap this maple tree." ) );
+        return;
+    }
+
+    if( !p.has_items_with_quality( "HAMMER", 1, 1 ) ) {
+        add_msg( m_info, _( "You need a tool to hammer the spile into the crust to tap this maple tree." ) );
+        return;
+    }
+
+    std::string spile_name = item( "maple_tree_spile", 0 ).tname( 1 );
+    std::string title = string_format( _( "Which %s:" ), spile_name.c_str() );
+
+    auto spile_loc = g->inv_map_splice( []( const item &it ) { return it.typeId() == "maple_tree_spile"; }, title, 1 );
```

Comment:
Then using a crafting inventory would be better. This one would have a range of 6 tiles instead of just 1 and wouldn't annoy the player with redundant menus when all spiles are from the same source.

Like this:
```
const inventory &crafting_inv = pl.crafting_inventory();
if( crafting_inv.has_amount( component_id, items_needed ) ) {
    std::vector<item_comp> comps;
    comps.push_back( item_comp( component_id, items_needed ) );
    pl.consume_items( comps );
}
```